docs/core-concepts/architecture/architecture-c4.puml

```diff
@@ -0,0 +1,40 @@
+@startuml hydra-components.png
+title "Hydra Components"
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
+
+AddElementTag("external", $bgColor="#23aa12")
+
+Container_Boundary(hydra_node, "Hydra Node") {
+   Component(persistence, "Persistence", "JSON/FS", "Persists state of the head on disk", $link="https://github.com/input-output-hk/hydra/blob/master/hydra-node/src/Hydra/Persistence.hs")
+   Component(head_logic, "Head Logic", "", "Maintains internal Head state and processes events", $link="https://github.com/input-output-hk/hydra/blob/master/hydra-node/src/Hydra/HeadLogic.hs")
+   Component(api_server, "API Server", "", "Dispatch client commands & queries and serves outputs", $link="https://github.com/input-output-hk/hydra/blob/master/hydra-node/src/Hydra/API/Server.hs")
+   Component(chain, "Direct Chain", "", "Connection to Cardano\nObserve txs on-chain and post tx to chain", $link="https://github.com/input-output-hk/hydra/blob/master/hydra-node/src/Hydra/Chain/Direct.hs")
```

Comment:
I think I would rather have this diagram (and the accompanying text) reflect the _actual architecture_, thus refer to  the current implementation, possibly even with alternatives if they exist, rather than some abstraction. Hence why the use of names like `Direct Chain` and `Tiny Wallet` (I agree the latter is not relevant anymore though).

Reply:
I finally followed your suggestion and represented things more abstractly.